R/missing_argument_linter.R

```diff
@@ -44,16 +44,16 @@ missing_argument_linter <- function(except = c("alist", "quote", "switch"), allo
 
   # require >3 children to exclude foo(), which is <expr><OP-LEFT-PAREN><OP-RIGHT-PAREN>
   xpath <- glue("
-    //SYMBOL_FUNCTION_CALL[not({ xp_text_in_table(except) })]
-      /parent::expr
+    parent::expr
       /parent::expr[count(*) > 3]
       /*[{xp_or(conds)}]
   ")
 
   Linter(linter_level = "file", function(source_expression) {
-    xml <- source_expression$full_xml_parsed_content
+    xml_targets <- source_expression$xml_find_function_calls(NULL, keep_names = TRUE)
```

Comment:
Maybe cleaner to have a parameter `drop=` or `exclude=` for `xml_find_function_calls()`?

Reply:
What would this parameter do?

Reply:
conjugate `function_names`, i.e. `function_names` sets which calls to _include_, this sets which calls to _exclude_. Then in this step we can do `$xml_find_function_calls(exclude = except)`, done.

We could also use it in `return_linter()`:

https://github.com/r-lib/lintr/blob/0f0b17c367e305ba2a3316f5cc532e1e2fa1e0c5/R/return_linter.R#L94

Reply:
Would file that for follow-up to keep focused, if that is okay with you.

Reply:
#2429

Reply:
#2430